File: server/common/messages/confirm_bet_batch.py
from ..interfaces.message import Message
from ..constants.message_codes import MessageCode
from ..interfaces.protocol import Protocol
from ..constants.result_codes import ResultCode


class ConfirmBetBatch(Message):
    def __init__(
        self,
        agency_id: int,
        bet_amount: int,
        result_code: ResultCode,
    ):
        self.agency_id = agency_id
        self.bet_amount = bet_amount
        self.result_code = result_code

    def get_message_code(self) -> MessageCode:
        return MessageCode.CONFIRM_BET_BATCH

    def send_to_lottery_central(self, lottery_central):
        raise NotImplementedError("ConfirmBetBatch is not sent to lottery central")

    def send_to_agency(self, protocol: Protocol):
        protocol.send_confirm_bet_batch(
            self.agency_id, self.bet_amount, self.result_code
        )

File: client/common/confirm_bet_batch.go
package common

type confirmBetBatch struct {
	agencyId   uint16
	betAmount  uint32
	resultCode ResultCode
}

func NewConfirmBetBatch(agencyId uint16, betAmount uint32, resultCode ResultCode) Message {
	return &confirmBetBatch{
		agencyId:   agencyId,
		betAmount:  betAmount,
		resultCode: resultCode,
	}
}

func (m *confirmBetBatch) GetMessageCode() MessageCode {
	return CONFIRM_BET_BATCH
}

func (m *confirmBetBatch) SendToLotteryCentral(protocol LotteryAgencyProtocol) error {
	// Implementation should be left empty
	return nil
}

func (m *confirmBetBatch) SendToAgency(agency LotteryAgency) error {
	return agency.ConfirmBetBatch(m.agencyId, m.betAmount, m.resultCode)
}
